pages/common/xml-list.md

```diff
@@ -0,0 +1,16 @@
+# xml list
+
+> List a directory's contents in XML format.
+> More information: <http://xmlstar.sourceforge.net/docs.php>.
+
+- List the current directory's contents to an XML document:
+
+`xml list >{{path/to/dir_list.xml}}`
+
+- List the specified directory's contents to an XML document:
+
+`xml list {{path/to/directory}} >{{path/to/dir_list.xml}}`
```

Comment:
```suggestion
> Print a directory's contents in XML format.
> More information: <http://xmlstar.sourceforge.net/docs.php>.

- Save the current directory's contents to an XML document:

`xml list >{{path/to/dir_list.xml}}`

- Save the specified directory's contents to an XML document:

`xml list {{path/to/directory}} >{{path/to/dir_list.xml}}`
```
Maybe `directory's structure` might be more fitting.

Reply:
I used "List" because that's the function of "ls" and this command. All of my examples "save" to a file because "printing" it to stdout isn't very useful. "ls" does that better.

Reply:
> List a directory's contents (like 'ls') in XML format.
> More information: <http://xmlstar.sourceforge.net/docs.php>.

- List the current directory's contents to an XML document:

Reply:
But when you for example pipe it to `lpr` you also wouldn't say `list to a printer`, but instead just `print`, even though the command on it's own lists something. I think `list to a file` doesn't sound right.

Reply:
The subcommand is "list". It lists the directory's contents, also known as a directory listing. No-one wants to use this command to "print" to a printer or even the screen. They want to write the directory's listing to a file or possibly pipe it to another xml command to operate on the XML-formatted data. "List" is the correct imperative verb for the "list" command. However, I'll compromise with you with "write" and use "listing" as a noun instead.
-----
> Write a directory's listing (like 'ls') in XML format.
> More information: <http://xmlstar.sourceforge.net/docs.php>.

- Write the current directory's listing to an XML document:

- Write the specified directory's listing to an XML document:

Reply:
I would recommend `display`.

with ls, the command literally means list directories and files. List makes sense.

Reply:
But it doesn't display anything @CleanMachine1.
@sbrl what do you think?

Reply:
Subcommand's internal help text. Note verb "list":
-----
```
❯ xml ls --help
XMLStarlet Toolkit: List directory as XML
Usage: xml ls [ <dir> | --help ]
Lists current directory in XML format.
Time is shown per ISO 8601 spec.
```
-----
Current compromise TLDR text (original used verb "list", like above):
-----
# xml list

> Write a directory's listing (like 'ls') in XML format.
> More information: <http://xmlstar.sourceforge.net/docs.php>.

- Write the current directory's listing to an XML document:

- Write the specified directory's listing to an XML document:
-----
Output of `xml ls` command (same as `xml list`):
-----
```
<dir>
<f p="rwxr-xr-x" a="20210808T151032Z" m="20210807T212608Z" s="3601"             n="tldr-lint-cli.js"/>
<f p="rw-r--r--" a="20210807T212608Z" m="20210807T212608Z" s="7051"             n="tldr-lint.js"/>
<f p="rw-r--r--" a="20210807T212609Z" m="20210807T212608Z" s="30021"            n="tldr-parser.js"/>
</dir>
```
Note that this is not a human-readable format and will not be "displayed" or "printed" for reading.
-----
Output of `ls` command, for comparison:
-----
```
-rwxr-xr-x 1 danielb staff  3601 Aug  7 17:26 tldr-lint-cli.js*
-rw-r--r-- 1 danielb staff  7051 Aug  7 17:26 tldr-lint.js
-rw-r--r-- 1 danielb staff 30021 Aug  7 17:26 tldr-parser.js
```

Reply:
I think I just realized that you're mainly talking about the first line in my suggestion. Then yes, `list` is fine if it's more correct. Sorry for not realizing what you where talking about 😅 
But for the others, I think `save` is still more accurete, because it saves it to a file.

Reply:
I'm sticking with "Write to a file". "Save" is for buffers.

Reply:
Sounds good